Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigation.m

```diff
@@ -0,0 +1,277 @@
+        [self.barView.leadingAnchor constraintEqualToAnchor:view.leadingAnchor constant:-BPKSpacingBase],
+        [self.barView.trailingAnchor constraintEqualToAnchor:view.trailingAnchor constant:BPKSpacingBase],
+    ];
+    [NSLayoutConstraint activateConstraints:self.barConstraints];
+}
+
+- (void)updateSelection:(UIButton *)sender {
+    NSInteger newIndex = [self.stackView.arrangedSubviews indexOfObject:sender];
+    [self setSelectedItemIndex:newIndex];
+
+    if ([self.delegate respondsToSelector:@selector(horizontalNavigation:didSelectItem:)]) {
+        [self.delegate horizontalNavigation:self didSelectItem:newIndex];
+    }
+}
+
+- (BPKHorizontalNavigationItem *)createHorizontalNavigationItemWithDefinition:
+    (BPKHorizontalNavigationOption *)definition {
+    return [[BPKHorizontalNavigationItem alloc] initWithDefinition:definition];
+}
+
+- (void)forEachNavigationItem:(void (^)(BPKHorizontalNavigationItem *))callback {
+    for (UIView *subView in self.stackView.arrangedSubviews) {
+        NSAssert([subView isKindOfClass:[BPKHorizontalNavigationItem class]],
+                 @"HorizontalNav subview is not of type BPKHorizontalNavigationItem as expected.");
+        if (![subView isKindOfClass:[BPKHorizontalNavigationItem class]]) {
+            continue;
+        }
+        callback((BPKHorizontalNavigationItem *)subView);
+    }
+}
+
+- (void)updateSelectedItemsColor {
+    [self forEachNavigationItem:^(BPKHorizontalNavigationItem *navigationItem) {
+      navigationItem.selectedColor = self.selectedColor;
+    }];
+}
+
+- (void)updateSelectedItemsFontMapping {
+    [self forEachNavigationItem:^(BPKHorizontalNavigationItem *navigationItem) {
+      navigationItem.fontMapping = self.fontMapping;
+    }];
+}
+
+- (void)setSelectedColor:(UIColor *_Nullable)selectedColor {
+    if (_selectedColor != selectedColor) {
+        _selectedColor = selectedColor;
+
+        [self updateBarColor];
+        [self updateSelectedItemsColor];
+    }
+}
+
+- (void)setFontMapping:(BPKFontMapping *_Nullable)fontMapping {
+    if (_fontMapping != fontMapping) {
+        _fontMapping = fontMapping;
+
+        [self updateSelectedItemsFontMapping];
+    }
+}
+
+#pragma mark - Private
+
+- (UIColor *_Nonnull)updateBarColor {
+    if (self.selectedColor != nil) {
+        return self.barView.backgroundColor = self.selectedColor;
+    } else {
+        return self.barView.backgroundColor = BPKColor.blue500;
+    }
+}
+
+- (void)setupWithOptions:(NSArray<BPKHorizontalNavigationOption *> *)options selected:(NSInteger)selectedItemIndex {
+
+    self.barView = [UIView new];
+    [self updateBarColor];
+    self.barView.translatesAutoresizingMaskIntoConstraints = NO;
+    [self addSubview:self.barView];
+
+    [NSLayoutConstraint activateConstraints:@[
+        [self.barView.topAnchor constraintEqualToAnchor:self.bottomAnchor constant:-BPKSpacingSm / 2],
+        [self.barView.leadingAnchor constraintGreaterThanOrEqualToAnchor:self.leadingAnchor],
```

Comment:
Can we keep the bar hidden while it's not being displayed so that these aren't required to resolve ambigious layout?